Assets/Plugins/iOS/Shopify/Checkout/Checkout.WebViewController.swift

```diff
@@ -116,4 +149,18 @@ extension Checkout.WebViewController: WKNavigationDelegate {
     private func isThankYouPage(_ url: URL) -> Bool {
         return url.absoluteString.hasSuffix("/thank_you")
     }
+    
+    private func handleLoadError(_ error: NSError) {
+        let errorPageURL: URL
+        switch error.code {
+        case NSURLErrorNotConnectedToInternet:
+            errorPageURL = generateErrorPage(message: "Looks like you don't have an internet connection. You won't be able to continue checking out until you do. Tap 'Reload' to try loading the page again.",
+                                             retryURL: lastGoodURL)
+        default:
+            errorPageURL =  generateErrorPage(message: "We've run into an unknown issue. Tap 'Reload' to try loading the page again.",
+                                              retryURL: lastGoodURL)
+        }
```

Comment:
Looks like the switch statement is deciding which `message` to use. Would it be cleaner if `generateErrorPage` was used after the `switch`, while the `switch` just picks a `message`